src/services/wallet/types.ts

```diff
@@ -7,12 +7,18 @@ export type Phrase = string
  * Type for providing 3 states of keystore
  *
  * (1) `None` -> DEFAULT (keystore needs to be imported at start of application or after shutdown of app)
- * (2) `Some<None>` -> LOCKED STATUS (keystore file, but no phrase)
- * (3) `Some<Some<KeystoreContent>>` -> UNLOCKED + IMPORTED STATUS (keystore file + phrase)
+ * (2) `Some<{status: LOCKED}>` -> LOCKED STATUS (keystore file, but no phrase)
+ * (3) `Some<{statue: UNLOCKED, phrase: Phrase}>` -> UNLOCKED + IMPORTED STATUS (keystore file + phrase)
  */
 
-export type KeystoreContent = { phrase: Phrase }
-export type KeystoreState = Option<Option<KeystoreContent>>
+export enum WalletStatus {
+  LOCKED = 'LOCKED',
+  UNLOCKED = 'UNLOCKED'
+}
+
+export type KeystoreContent = { status: WalletStatus.LOCKED } | { status: WalletStatus.UNLOCKED; phrase: Phrase }
+
+export type KeystoreState = Option<KeystoreContent>
```

Comment:
As mentioned in Slack, I'm not happy about this change: More code, it includes redundant information. IMO we don't need this change.

Reply:
We've decided to keep this change ^